src/Bindings/Browser/ObjectService.php

```diff
@@ -941,4 +1017,75 @@ public function updateProperties(
             }
         }
     }
+
+    /**
+     * @param string $identifier
+     * @param mixed $additionalHashValues
+     * @return array
+     */
+    protected function createCacheKey($identifier, $additionalHashValues)
+    {
+        return array(
+            $identifier,
+            sha1(is_array($additionalHashValues) ? serialize($additionalHashValues) : $additionalHashValues)
```

Comment:
The problem i see here is that the cache key depends on the ordering of additionalHashValues. What do you think about adding a special cache object? Maybe with a magic getter and setter even if that is... well... ugly...
But i think we need a consistend way to create the cache key...
We alredy talked about implementing a 3rd party cache framework which really would make sense for the whole lib but yes we have really specific needs...
Any comments on that?